### ISSUE TITLE
cdef_dist_wxh() is called for 4x4 blocks

### ISSUE THREAD

**Issue by ycho** [xiph/rav1e#432]
PR #431 adds asserts to detect this bug.

**Comment by ycho** [comment]
Options to resolve this bug can be:
1) Just don't use 4x4 blocks or bsizes < 8x8, so set fi.min_partition_size = 8x8.
This would be maybe a temporary solution but at least until real fix lands.
2) Fix, so that for 4x4 blocks, each 4x4 uses MSE for mode decision then apply cdef_dist_8x8 on 8x8 pixels consiting of 4x4 blocks.
(and optionally revert the patch or take option 1) above until fixed.)

**Comment by ycho** [comment]
Landed temporary fix:
 https://github.com/xiph/rav1e/pull/433/commits/68b0aa2422378a7afe94a27361cb63d42d4b3481

**Comment by ycho** [comment]
If no one want to fix this, let's close this.
My opinion is not to proceed further.

**Comment by tdaede** [comment]
We could run MSE on 4x4 blocks.

**Comment by ycho** [comment]
Turns out to me that was rocket science in libaom, and I suggest not doing that in rav1e.
If anyone want to try, I'd not opposed though.

**Comment by ycho** [comment]
For anyone interested in tackle on this issue, i.e. using MSEs for the mode decision of sub8x8 blocks, then applying cdef-dist on the 8x8 block as we have tried on libaom, I'd like to recommend that try the idea **after** you find enough evidences that cdef-dist with "min partition size 8x8" encoding contributes to the quality improvement over current default MSE (so, current --speed=2 in rav1e).
I wish I had proposed this way for dist_8x8 from the beginning.
